stdlib/public/core/EmbeddedRuntime.swift

```diff
@@ -196,23 +196,31 @@ fileprivate func refcountPointer(for object: UnsafeMutablePointer<HeapObject>) -
   return UnsafeMutablePointer<Int>(UnsafeRawPointer(object).advanced(by: MemoryLayout<Int>.size)._rawValue)
 }
 
-fileprivate func loadRelaxed(_ refcount: UnsafeMutablePointer<Int>) -> Int {
-  Int(Builtin.atomicload_monotonic_Word(refcount._rawValue))
+fileprivate func loadRelaxed(_ atomic: UnsafeMutablePointer<Int>) -> Int {
+  Int(Builtin.atomicload_monotonic_Word(atomic._rawValue))
 }
 
-fileprivate func loadAcquire(_ refcount: UnsafeMutablePointer<Int>) -> Int {
-  Int(Builtin.atomicload_acquire_Word(refcount._rawValue))
+fileprivate func loadAcquire(_ atomic: UnsafeMutablePointer<Int>) -> Int {
+  Int(Builtin.atomicload_acquire_Word(atomic._rawValue))
 }
 
-fileprivate func subFetchAcquireRelease(_ refcount: UnsafeMutablePointer<Int>, n: Int) -> Int {
-  let oldValue = Int(Builtin.atomicrmw_sub_acqrel_Word(refcount._rawValue, n._builtinWordValue))
+fileprivate func subFetchAcquireRelease(_ atomic: UnsafeMutablePointer<Int>, n: Int) -> Int {
+  let oldValue = Int(Builtin.atomicrmw_sub_acqrel_Word(atomic._rawValue, n._builtinWordValue))
   return oldValue - n
 }
 
-fileprivate func addRelaxed(_ refcount: UnsafeMutablePointer<Int>, n: Int) {
-  _ = Builtin.atomicrmw_add_monotonic_Word(refcount._rawValue, n._builtinWordValue)
+fileprivate func addRelaxed(_ atomic: UnsafeMutablePointer<Int>, n: Int) {
+  _ = Builtin.atomicrmw_add_monotonic_Word(atomic._rawValue, n._builtinWordValue)
 }
 
+fileprivate func compareExchangeRelaxed(_ atomic: UnsafeMutablePointer<Int>, expectedOldValue: Int, desiredNewValue: Int) -> Bool {
```

Comment:
we need to be careful with these: Cortex-M 0 don't have a lot when it comes to atomics. From my understanding all they have are DMB and register based spin locks (which the spin locks are not accessible by compiler intrinsics, and require library level access to those).

Reply:
Good point. Though these atomic builtins are valid LLVM builtins for *any* target and it's the LLVM backend's job to lower them to something that works, worst case a library call. I think that we should (1) keep using atomic builtins here in the embedded runtime code and rely on compiler-rt or a user-provided library to make them actually work, and if the user knows that the system is single-threaded, those can be implemented in a trivial way, and (2) eventually provide an option to use an (alternative) single-threaded embedded runtime which doesn't do any of the atomic stuff (and saves a little bit of code of on that).